### PR TITLE
[16.0][MIG] project_timesheet_time_control: Migration to version 16.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc @Tecnativa TT43239

@Yadier-Tecnativa @CarlosRoca13 please review!

this PR superseeds #1133

### CODE REVIEW

**Review by Yadier-Tecnativa — APPROVED** [review]
the code LGTM

**Comment by ygol** [comment]
getting error on runbot
1. click on "Start work" in a task
2. add a description
3. click on "Start New Timer"
4. getting following error
```
Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1584, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1611, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 1815, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 697, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 48, in call_button
    return clean_action(action, env=request.env)
  File "/opt/odoo/addons/web/controllers/utils.py", line 35, in clean_action
    readable_fields = env[action['type']]._get_readable_fields()
  File "/opt/odoo/odoo/api.py", line 537, in __getitem__
    return self.registry[model_name](self, (), ())
  File "/opt/odoo/odoo/modules/registry.py", line 190, in __getitem__
    return self.models[model_name]
KeyError: 'ir.actions.act_multi'

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (http://oca-project-16-0-pr1148-5eae6387e809.runboat.odoo-community.org/web/assets/627-1bdd447/web.assets_backend.min.js:985:163)
    at XMLHttpRequest.<anonymous> (http://oca-project-16-0-pr1148-5eae6387e809.runboat.odoo-community.org/web/assets/627-1bdd447/web.assets_backend.min.js:993:13)
```

**Comment by pedrobaeza** [comment]
/ocabot migration project_timesheet_time_control

@stefan-tecnativa please remove the code that is doing reference to the removed dependencies, like the one with the action `ir.actions.act_multi`.

**Comment by stefan-tecnativa** [comment]
> /ocabot migration project_timesheet_time_control
> 
> @stefan-tecnativa please remove the code that is doing reference to the removed dependencies, like the one with the action `ir.actions.act_multi`.

**Comment by stefan-tecnativa** [comment]
I closed it by mistake 😅
@ygol corrected, you can try again now.

File: project_timesheet_time_control/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+    "name": "Project timesheet time control",
+    "version": "16.0.1.0.0",
+    "category": "Project",
+    "author": "Tecnativa," "Odoo Community Association (OCA)",
+    "maintainers": ["ernestotejeda"],
+    "website": "https://github.com/OCA/project",
+    "depends": [
+        "hr_timesheet_task_domain",
+        "hr_timesheet_task_stage",
```
Comment (pedrobaeza) [inline project_timesheet_time_control/__manifest__.py]: I think we can remove these dependencies as well as just let `hr_timesheet`, as they are not strictly needed.

**Comment by ygol** [comment]
> I closed it by mistake sweat_smile @ygol corrected, you can try again now.

Not getting the error anymore.

**Review by AntoniRomera — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]
Code and functional review :+1:

**Review by DorianMAG — APPROVED** [review]
Hi @stefan-tecnativa.
Thx for this work !
Tested on ruboat.
LGTM

File: project_timesheet_time_control/views/account_analytic_line_view.xml [hunk 2]
```diff
@@ -37,31 +37,30 @@
...
                 <button
-                    name="button_resume_work"
-                    type="object"
-                    icon="fa-play-circle text-success"
-                    attrs="{'invisible': [('show_time_control', '!=', 'resume')]}"
-                    class="oe_stat_button"
-                    context="{'show_created_timer': True}"
-                    title="Resume work"
-                />
+                        name="button_resume_work"
```
Comment (pedrobaeza) [inline project_timesheet_time_control/views/account_analytic_line_view.xml:46]: This indentation change is not correct. Please let it as it was.

File: project_timesheet_time_control/views/account_analytic_line_view.xml [hunk 3]
```diff
@@ -37,31 +37,30 @@
...
                 <button
-                    name="button_end_work"
-                    type="object"
-                    icon="fa-stop-circle text-warning"
-                    attrs="{'invisible': [('show_time_control', '!=', 'stop')]}"
-                    class="oe_stat_button"
-                    title="Stop work"
-                />
-            </div>
+                        name="button_end_work"
```
Comment (pedrobaeza) [inline project_timesheet_time_control/views/account_analytic_line_view.xml:55]: The same about indentation.

File: project_timesheet_time_control/views/project_project_view.xml [hunk 4]
```diff
@@ -4,10 +4,6 @@
         <field name="name">Add timesheet time controls</field>
         <field name="model">project.project</field>
         <field name="inherit_id" ref="hr_timesheet.project_invoice_form" />
-        <field
-            name="groups_id"
-            eval="[(6, 0, [ref('hr_timesheet.group_hr_timesheet_user')])]"
```
Comment (pedrobaeza) [inline project_timesheet_time_control/views/project_project_view.xml]: If you remove these groups from the view, you have to put them at field/page/group element level.

File: project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml [hunk 5]
```diff
@@ -23,10 +23,10 @@
                         <field name="project_id" required="1" />
                         <field name="task_id" />
                         <field name="company_id" groups="base.group_multi_company" />
+                        <field name="company_id" invisible="1" />
                     </group>
-                    <group
+                    <div
                         name="messages"
-                        string="Previous timer"
```
Comment (pedrobaeza) [inline project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml]: You are removing a text that was appearing here (`Previous timer`). Please put the equivalent one, but you can't remove it.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml (hunk 5 above)
Comment (stefan-tecnativa) [reply project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml]: Changes corrected, thanks!

**Comment by AntoniRomera** [comment]
@pedrobaeza Is this already correct?

File: project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml (hunk 5 above)
Comment (pedrobaeza) [reply project_timesheet_time_control/wizards/hr_timesheet_switch_view.xml]: It was not correct. The label was still not shown, but even more, it has problems not showing in the same line the rest of the fields. Now it is.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request closed without merging**